scene/animation/animation_blend_tree.cpp

```diff
@@ -1459,6 +1467,7 @@ void AnimationNodeBlendTree::add_node(const StringName &p_name, Ref<AnimationNod
 	n.position = p_position;
 	n.connections.resize(n.node->get_input_count());
 	nodes[p_name] = n;
+	p_node->set_name(p_name);
```

Comment:
```suggestion
```

This should never be done. It is the AnimationTree's role to manage the names of AnimationNodes, and AnimationNodes should not contain names.

As explained in some places like https://github.com/godotengine/godot/pull/101564#pullrequestreview-2561174593, an AnimationNode is a resource, so we need to consider the case where multiple AnimationNodes with the same reference are placed in the one blend tree.

Reply:
I've replaced this with a node to name map which is maintained in parallel to nodes as described in the previous comment